src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -1,5 +1,62 @@
 import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute } from '@angular/router';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable, of } from 'rxjs/index';
+import { map } from 'rxjs/internal/operators';
+import { HttpRequestService } from '../../../services/http-request.service';
+import { StatusMessageService } from '../../../services/status-message.service';
+import { ErrorMessageOutput } from '../../error-message-output';
+import { StudentListSectionData, StudentListStudentData } from '../student-list/student-list-section-data';
+
+interface Course {
+  id: string;
+  name: string;
+  isArchived: boolean;
+  isInstructorAllowedToModify: boolean;
+  isChecked: boolean | undefined;
```

Comment:
Why do all of these need to be `| undefined`?

Reply:
this is because this parameter is not passed from the backend, and yet i need it to be part of this object...

Reply:
Then you're looking for `isChecked?: boolean`.